### PR TITLE
Chaklakl/sprint2 code review

### PR DESCRIPTION
**Pull request by ckenil**
I think this group has made a lot of meaningful progress from the last sprint iteration. The new features that I saw were consistent with the design patterns we have seen in class which was nice to see. The structure of the project continues to follow good standards in that everything was properly structured for easy access and navigation. I was able to get your code to compile and see the new pages you have added. Although these pages are lacking in functionality, I can see the approach you are going for in accordance with the structure of the back-end. Overall, I believe you guys have done a great job since the last sprint and the comments have helped me see your current and future progress!

The program was available on time and I was able to compile it.

Three concepts I learned through your code:

1. From your profile.html class, I saw how you guys were creating links for your Profile objects which is something that I haven't done before. By utilizing the <a> tag and your controller, I was able to see how to create links for profile objects.
2. I learned the concept of displaying data entities to your front-end from how you were displaying the list of events from the stubbed objects. This syntax: th:each="event : ${events}", is still relatively new to me, so seeing it in your code and playing around with it made it easier to learn.
3. Although this is something that has been shown in the lectures, I learned more about the MVC framework with how you handled the models within your controller to get them to display to views. Since everything was well documented, I was able to follow along through how everything was interacting together.

### CODE REVIEW

File: src/main/java/com/trailfinder/TrailFinderController.java [hunk 1]
```diff
@@ -23,7 +23,6 @@
 @Controller
 public class TrailFinderController {
 	
-	// Private Fields
```
Comment (ckenil) [inline src/main/java/com/trailfinder/TrailFinderController.java:26]: Removed unnecessary comment

File: src/main/java/com/trailfinder/dto/EventDTO.java [hunk 2]
```diff
@@ -22,6 +21,38 @@
 	private ProfileDTO eventCreator;
 	private List<ProfileDTO> attendees;
 	
+	/**
```
Comment (ckenil) [inline src/main/java/com/trailfinder/dto/EventDTO.java:24]: Overloaded constructor for initializing DTO with all fields during instantiation. Not sure which fields were vital, but you could overload more constructors if I included non-essential fields in this constructor.

File: src/main/java/com/trailfinder/dto/EventDTO.java [hunk 3]
```diff
@@ -22,6 +21,38 @@
...
+		this.Latitude = latitude;
+		this.Longitude = longitude;
+		this.eventCreator = eventCreator;
+		this.attendees = attendees;
+	}
+	
+	/**
+	 * Create an instance of EventDTO
+	 */
+	public EventDTO() {
```
Comment (ckenil) [inline src/main/java/com/trailfinder/dto/EventDTO.java:52]: Created default constructor for when you don't want to initialize fields during instantiation.

File: src/main/java/com/trailfinder/service/HikeServiceStub.java [hunk 4]
```diff
@@ -55,13 +55,7 @@ public void saveProfile(ProfileDTO profile) {
...
 			ProfileDTO profile = new ProfileDTO(i, "test " + i, "TestBio " + i, null);
 			
-			EventDTO event = new EventDTO();
-			event.setEventId(i);
-			event.setLatitude(39.1330 + i);
-			event.setLongitude(-84.4134 + i);
-			event.setDistance(i + (i/2));
-			event.setEventCreator(profile);
-			events.add(event);
+			events.add(new EventDTO(i, 39.1330 + i, null, null, -84.4134 + i, i + (i/2), profile, null));
```
Comment (ckenil) [inline src/main/java/com/trailfinder/service/HikeServiceStub.java:58]: Used overloaded constructor to initialize new EventDTOs and add them to events.

File: src/main/java/com/trailfinder/service/IHikeService.java [hunk 5]
```diff
@@ -16,31 +16,31 @@
 public interface IHikeService {
 
 	/*
```
Comment (ckenil) [inline src/main/java/com/trailfinder/service/IHikeService.java:18]: Removed 'shoulds' from comments as the functionality of the methods are implied.

File: src/main/java/com/trailfinder/TrailFinderController.java (hunk 1 above)
Comment (discospiff) [reply src/main/java/com/trailfinder/TrailFinderController.java:26]: Agreed.  You can determine that by looking at the access modifier.

File: src/main/java/com/trailfinder/dto/EventDTO.java (hunk 3 above)
Comment (discospiff) [reply src/main/java/com/trailfinder/dto/EventDTO.java:52]: These days, many classes only have a default constructor.  Unless:
1) There are attributes that are required for the object to be constructed... like a database connection.
2) You are using constructor dependency injection, in Spring, or 
3) You're using Kotlin, where attributes tend to be initialized as constructor parameters.

File: src/main/java/com/trailfinder/service/HikeServiceStub.java (hunk 4 above)
Comment (discospiff) [reply src/main/java/com/trailfinder/service/HikeServiceStub.java:58]: I'd leave this as it were, for documentation purposes.  Reason being, the setters are more descriptive, where you have to make assumptions on what "null, null, i + (i/2)" means.

File: src/main/java/com/trailfinder/service/IHikeService.java (hunk 5 above)
Comment (discospiff) [reply src/main/java/com/trailfinder/service/IHikeService.java:18]: Agreed.

**Comment by discospiff** [comment]
Several good suggestions here. There are a few others that I will leave up to the group to decide.  See inline comments for details.

**Pull request merged**